src/Calculator/Utils/Delayer.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "pch.h"
+
+namespace CalculatorApp
+{
+public
+    ref class Delayer sealed
+    {
+    public:
+        Delayer(Windows::Foundation::TimeSpan timeSpan);
+
+        void Start();
+        void ResetAndStart();
+        void Stop();
+
+        event Windows::Foundation::EventHandler<Platform::Object ^> ^ Action;
+
+     private:
+        void Timer_Tick(Platform::Object ^ sender, Platform::Object ^ e);
+
+        Windows::UI::Xaml::DispatcherTimer ^ m_timer;
```

Comment:
Is this the right timer to use? Probably you don't care which thread the action fires on, so a threadpool timer would be a better choice. Also, can we avoid creating one timer per slider and instead use one per instance of EquationInputArea?

Reply:
Right now a threadpool timer works as well since the current logic doesn't access a xaml object. However, in the future, we'll be able to use the Delayer to access xaml objects if we leave the DispatcherTimer instead of updating it to a threadpool timer.

For the number of timers, we want to be able to log when each variable is used, so we want them to have their own timers. I have updated the event to log the variable name as well.

Reply:
DispatcherTimers come with significant overhead; we shouldn't pay for that until we actually need it.

Reply:
I would agree with you in general, but not especially in this particular case because we stop/restart the timer a lot (ThreadpoolTimer does not provide a way to reset a timer and you are forced to create a new timer, which will create a huge overload, especially when associated with a Slider that will spam events) and because the action is not a time-consuming task.

Reply:
Resetting the dispatchertimer will probably do just as much work as creating a new threadpool timer. I agree that the API is harder to use, though. So I guess I'm fine keeping this as written, for this particular instance.

The reason I'm being picky here is that I have tried and failed to write "generic"/"reusable" timer helpers before. They tend to become timebombs--someone else inevitably comes along and uses it without understanding the tradeoffs the original author made and gets into trouble. Maybe naming this class "DispatcherTimerDelayer" would help discourage that outcome.

Reply:
I renamed the class to DispatcherTimerDelayer